jib-core/src/test/java/com/google/cloud/tools/jib/builder/steps/PlatformCheckerTest.java

```diff
@@ -58,39 +58,43 @@ public void testCheckManifestPlatform_mismatch() {
     ContainerConfigurationTemplate containerConfigJson = new ContainerConfigurationTemplate();
     containerConfigJson.setArchitecture("actual arch");
     containerConfigJson.setOs("actual OS");
-
-    PlatformChecker.checkManifestPlatform(buildContext, containerConfigJson);
-
-    Mockito.verify(eventHandlers)
-        .dispatch(
-            LogEvent.warn(
-                "the configured platform (configured arch/configured OS) doesn't match the "
-                    + "platform (actual arch/actual OS) of the base image (scratch)"));
+    Exception ex =
+        assertThrows(
+            PlatformNotFoundInBaseImageException.class,
+            () -> PlatformChecker.checkManifestPlatform(buildContext, containerConfigJson));
+    assertThat(ex)
+        .hasMessageThat()
+        .isEqualTo(
+            "the configured platform (configured arch/configured OS) doesn't match the "
+                + "platform (actual arch/actual OS) of the base image (scratch)");
   }
 
   @Test
-  public void testCheckManifestPlatform_noWarningIfDefaultAmd64Linux() {
+  public void testCheckManifestPlatform_noWarningIfDefaultAmd64Linux()
+      throws PlatformNotFoundInBaseImageException {
```

Comment:
Oops sorry I missed this earlier. This test no longer performs an assertion so just invoking the method in the test may not give us many insights into what is expected from the method's behavior. How about we replace this test with the following?: `testCheckManifestPlatform_arm64Linux` where the architecture doesn't match but the os matches. This will help us test the `&&` in the following condition: `!(platform.getArchitecture().equals("amd64") && platform.getOs().equals("linux"))`

Reply:
> just invoking the method in the test may not give us many insights into what is expected from the method's behavior

On the contrary I think that the passing test implies the expectation that _it should not throw any exception_. Also this [yaqs](https://yaqs.corp.google.com/eng/q/5403980279709696#a6384374680518656) answer recommends that.